server/src/main/java/org/apache/druid/server/coordinator/compact/NewestSegmentFirstIterator.java

```diff
@@ -339,6 +339,12 @@ private SegmentsToCompact findSegmentsToCompact(
     while (compactibleSegmentIterator.hasNext()) {
       List<DataSegment> segments = compactibleSegmentIterator.next();
 
+      // Do not compact an interval which comprises of a single tombstone
+      // If there are multiple tombstones in the interval, we may still want to compact them
+      if (segments.size() == 1 && segments.get(0).isTombstone()) {
+        continue;
```

Comment:
Maybe add a warn log here.

Reply:
The logic should probably be moved to `isCompactibleHolder`.

Reply:
https://github.com/apache/druid/blob/e49a7bb3cd97764942dd2b465b0f94dc40653cef/server/src/main/java/org/apache/druid/server/coordinator/compact/NewestSegmentFirstIterator.java#L133

The tombstones are also considered to be numbered partition chunks in `isCompactibleHolder` because of the above code, and fixing it there seems a bit more involved.

Reply:
I am not sure I follow. You can still do `PartitionChunk.getObject` to access the `DataSegment` object,  right?

Reply:
Yes, but a DataSegment returned that way will return false when checking if it is a tombstone.
I have also made the change locally and the test fails